### DISCUSSION TITLE
ModemManager gives “Invalid transition” on EM7455

### DISCUSSION THREAD

**Discussion by gnull** [danielewood/sierra-wireless-modems#122] [Q&A] [answered]
I bought a “01AX748 Sierra AirPrime EM7455 4G/LTE WWAN” on Ebay and I'm tryuing to connect it to my Thinkpad X270. After installing it, I enabled ModemManager and tried doing:

```
$ mmcli -v -m 0 --simple-connect=apn=internet.telenor.se
[14 May 2022, 04:53:59] [Debug] Forcing request to be run asynchronously
[14 May 2022, 04:53:59] [Debug] Assuming '0' is the modem index
[14 May 2022, 04:53:59] [Debug] ModemManager process found at ':1.5'
[14 May 2022, 04:53:59] [Debug] Modem found at '/org/freedesktop/ModemManager1/Modem/0'

[14 May 2022, 04:53:59] [Debug] Asynchronously connecting the modem...
error: couldn't connect the modem: 'GDBus.Error:org.freedesktop.ModemManager1.Error.Core.Retry: Invalid transition'
```

It keeps giving me the “Invalid transition” error. If I try to enable the modem, I get the same error. Does anyone know what could be causing it?

It works fine if I'm trying to get the modem info:

```
$ mmcli -v -m 0
[14 May 2022, 04:54:45] [Debug] ModemManager process found at ':1.5'
[14 May 2022, 04:54:45] [Debug] Assuming '0' is the modem index
[14 May 2022, 04:54:45] [Debug] Modem found at '/org/freedesktop/ModemManager1/Modem/0'

[14 May 2022, 04:54:45] [Debug] Printing modem info...
  --------------------------------
  General  |                 path: /org/freedesktop/ModemManager1/Modem/0
           |            device id: <censored>
  --------------------------------
  Hardware |         manufacturer: Sierra Wireless, Incorporated
           |                model: Sierra Wireless EM7455 Qualcomm Snapdragon X7 LTE-A
           |    firmware revision: SWI9X30C_02.33.03.00
           |       carrier config: default
           |         h/w revision: EM7455
           |            supported: gsm-umts, lte
           |              current: gsm-umts, lte
           |         equipment id: <censored>
  --------------------------------
  System   |               device: /sys/devices/pci0000:00/0000:00:14.0/usb1/1-3
           |              drivers: cdc_mbim, qcserial
           |               plugin: sierra
           |         primary port: cdc-wdm0
           |                ports: cdc-wdm0 (mbim), ttyUSB0 (qcdm), ttyUSB2 (at), usb0 (net)
  --------------------------------
  Status   |       unlock retries: sim-pin2 (3)
           |                state: disabled
           |          power state: low
           |       signal quality: 0% (cached)
  --------------------------------
  Modes    |            supported: allowed: 3g; preferred: none
           |                       allowed: 4g; preferred: none
           |                       allowed: 3g, 4g; preferred: 4g
           |                       allowed: 3g, 4g; preferred: 3g
           |              current: allowed: 3g, 4g; preferred: 4g
  --------------------------------
  Bands    |            supported: utran-1, utran-3, utran-4, utran-5, utran-8, utran-2,
           |                       eutran-1, eutran-2, eutran-3, eutran-4, eutran-5, eutran-7, eutran-8,
           |                       eutran-12, eutran-13, eutran-20, eutran-25, eutran-26, eutran-41
           |              current: utran-1, utran-3, utran-4, utran-5, utran-8, utran-2,
           |                       eutran-1, eutran-2, eutran-3, eutran-4, eutran-5, eutran-7, eutran-8,
           |                       eutran-12, eutran-13, eutran-20, eutran-25, eutran-26, eutran-41
  --------------------------------
  IP       |            supported: ipv4, ipv6, ipv4v6
  --------------------------------
  3GPP     |                 imei: <censored>
           |        enabled locks: fixed-dialing
  --------------------------------
  3GPP EPS | ue mode of operation: csps-2
  --------------------------------
  SIM      |     primary sim path: /org/freedesktop/ModemManager1/SIM/0
```

https://gist.github.com/gnull/bc3ba58f4159984be56279e7d28ecdb9

Here's what the debug output of ModemManager looks like when I run the simple connect command. Any ideas what might be causing the problem?

**Comment by artilerien** [comment]
Did you come up with any solution to this? I'm having the exact same issue on my Dell Latitude laptop.

**Reply by gnull** [reply]
The new modem I bought works flawlessly. It must have been a compatibility problem (or they sold me a non-functioning modem).

To clarify. My laptop is Thinkpad X270. The first modem I bought was advertised as compatible with X260, it didn't work on my laptop. The second modem I got was advertised as compatible with X270, and it works just perfect. Both are Sierra Wireless EM7455 FRU 01AX748.

Hope this helps. I'd love to know more details of what exactly causes the issue, so if someone knows anything please post it here.

**Comment by gnull** [comment]
Nope. My current hypothesis is that my lte card might be incompatible with
my laptop. The ad from which I bought it stated that it's for X260 and my
laptop is X270.

I've ordered a different lte card. Will check if it works with it in a week
or so, and let you know.

Ivan

**Reply by gnull** [reply]
The new card works. So the issue must be caused by the first card being incompatible.

**Comment by Hetti** [comment, marked as answer]
I had the same problem and did more research.

I got it working. The thing is you need to tell the card the FCC Unlock command!
`qmicli -d /dev/cdc-wdm0 -p --dms-set-fcc-authentication`
See the quoted post: https://lists.freedesktop.org/archives/libqmi-devel/2020-October/003400.html

I got it working afterwards

**Reply by 5eb38fd8cdee** [reply]
This fixed my issue. Thank you!

**Reply by Hetti** [reply]
hey @gnull could you mark my question as correct answer? thanks =D

**Reply by gnull** [reply]
I haven't confirmed it myself, but I trust you guys when you say it helped.

**Comment by gnull** [comment]
Thank you for an update. Unfortunately, I can't test it on the modem
where I had the issue since I sold it already :)

Ivan
